### PR TITLE
Feat/local models

### LINKED ISSUE #18: Support evaluation of local PyTorch models
This should include the following:

1. Allow specifying the (relative) path to a local `bin` file with the model, input the `torch.nn.Module` instance itself, if the scripting approach to the package is used.
2. Useful error messages if various things could not be handled.
3. Some extended logging (if `verbose` is enabled), displaying inferred elements of the model, such as label enumeration, output type (logits/probabilities) etc.

### PR DESCRIPTION
**Pull request by saattrupdan**
This PR includes support for local PyTorch models, with the relevant functions located in the `local_model_utils` module.

This assumes that there is a "model folder" where the following are located:
- A Python file which defines the model architecture, as a subclass of `torch.nn.Module`.
- A `.bin` file with the model weights.

A configuration JSON file will be generated in the folder if the doesn't already exist, and the user will be prompted to enter the missing details. This includes:
- `tokenizer_id`: The Hugging Face ID to the tokenizer.
- `id2label`: The labels the model was trained on, in the correct order.
- Any arguments needed in the model initializer, which doesn't have any default values. If they don't have any type hints either then an informative error is shown.

A substantial change is also that I catch _all_ errors and log them using `log.error`, rather than using Python's own error logging. This for instance results in the more friendly:
```
2022-09-15 13:41:31,969 [INFO] <aiai_eval.evaluator>
↳ Evaluating the attack model on the offensive text classification task.
2022-09-15 13:41:35,412 [ERROR] <aiai_eval.evaluator>
↳ ValueError: The model architecture file attack/model.py does not exist.
```

A full traceback can be outputted if `verbose` is set.

This closes #18

### CODE REVIEW

**Comment by AJDERS** [comment]
I think we need to added a subsection of the Quickstart in the readme which details how to evaluate a local model.

File: src/aiai_eval/local_model_utils.py [hunk 1]
```diff
@@ -0,0 +1,375 @@
...
+        config = dict()
+    else:
+        config_path = config_paths[0]
+        config = json.loads(config_path.read_text())
+
+    # If the key is not in the config then we either use the default value or prompt
+    # the user to enter it
+    if key not in config:
+
+        # If the default value is set and is of the correct type, then we use it
```
Comment (AJDERS) [inline src/aiai_eval/local_model_utils.py:315]: Maybe wrap the contents of this loop in a while loop, and break it once we have received valid input, something like:

```
for key not in config:
    user_gave_invalid_value = True
    while user_gave_invalid_value:
        ...
        [Do bunch of stuff]
        ...
        try:
            else:
                if expected_type is str:
                    config[key] = user_input
                elif expected_type is int:
                    config[key] = int(user_input)
                elif expected_type is float:
                    config[key] = float(user_input)
                elif expected_type is bool:
                    config[key] = user_input.lower() == "true"
                elif expected_type is list:
                    config[key] = user_input.split()
                elif expected_type is dict:
                    config[key] = dict(
                        item.split("=") for item in user_input.split(",")
                    )
                user_gave_invalid_value = False
        except ValueError as e:
            [Tell people "hallo" is not an valid int value... and they should try again]
```

Right now we are assuming users can provide correct input, which is not true :(

File: src/aiai_eval/model_adjustment.py [hunk 2]
```diff
@@ -136,16 +103,14 @@ def adjust_model_to_task(
...
         # Get the old model id2label conversion
-        old_model_id2label = [
-            model.config.id2label[idx].upper()
-            for idx in range(len(model.config.id2label))
-        ]
+        old_model_id2label = model_config.id2label
 
         # Alter the model's classification layer to match the dataset if the model is
-        # missing labels
+        # missing labels. This only works if the model is a Hugging Face PyTorch model
```
Comment (AJDERS) [inline src/aiai_eval/model_adjustment.py:109]: I opened an issue addressing this point: #73 . For now we could log a snippet at this point that one should be wary with local models?

File: src/aiai_eval/local_model_utils.py (hunk 1 above)
Comment (AJDERS) [reply src/aiai_eval/local_model_utils.py:315]: I just realized i am mixing try/except and a if/else blocks in the above, but you get my gist :)

Comment (saattrupdan) [reply src/aiai_eval/local_model_utils.py:315]: Makes sense! Coming up 🙂

File: src/aiai_eval/model_adjustment.py (hunk 2 above)
Comment (saattrupdan) [reply src/aiai_eval/model_adjustment.py:109]: Hmm. If a user uses a local model then they'd simply encounter an error if the model doesn't have the correct amount labels. I suppose I can add an informative error if they supply fewer labels than what is in the dataset?

**Comment by saattrupdan** [comment]
> I think we need to added a subsection of the Quickstart in the readme which details how to evaluate a local model. 

Good point! I created an issue with that bit.

**Review by AJDERS — APPROVED** [review]
LGTM!

**Pull request merged**